src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -1033,6 +1039,21 @@ public StringBuilder Append(char[] value)
             return this;
         }
 
+        public StringBuilder Append(ReadOnlySpan<char> value)
+        {
+            if (value.Length > 0)
+            {
+                unsafe
+                {
+                    fixed (char* valueChars = &value.DangerousGetPinnableReference())
```

Comment:
Let's keep such things separate.  There are two distinct tracks for span in coreclr/corefx:
- Using it internally to clean up duplicated code, to make existing unsafe code safer, to make it easier to stack allocate in places we're currently heap allocating, etc.
- Adding new APIs that accept span

I'd like to largely keep those separate.  If it's easy to achieve the former while doing the latter, great, but doing so also makes it more likely we'll incur performance regressions, especially at this early stage where we haven't shaken out all of the possible perf issues with span.